src/Core.UnitTests/GitEventsMonitorTests.cs

```diff
@@ -0,0 +1,104 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace SonarLint.VisualStudio.Core.UnitTests
+{
+    [TestClass]
+    public class GitEventsMonitorTests
+    {
+        [TestMethod]
+        public void HeadChanged_EventInvoked()
+        {
+            int counter = 0;
+
+            var fileSystemWatcher = new Mock<IFileSystemWatcher>();
+
+            var fileSystemWatcherFactory = new Mock<IFileSystemWatcherFactory>();
+            fileSystemWatcherFactory.Setup(f => f.FromPath("C:\\Some Path\\.git")).Returns(fileSystemWatcher.Object);
+
+            var testSubject = CreateTestSubject("C:\\Some Path", fileSystemWatcherFactory);
+
+            fileSystemWatcher.VerifySet(w => w.Filter = "HEAD", Times.Once);
+            fileSystemWatcher.VerifySet(w => w.EnableRaisingEvents = true, Times.Once);
+
+            testSubject.HeadChanged += (o, e) => counter++;
+
+            fileSystemWatcher.Raise(w => w.Changed += null, null, null);
+
+            counter.Should().Be(1);
+        }
+
+        [TestMethod]
+        public void HeadChanged_MonitorDisposed_FileSystemWatcherDisposed()
+        {
+            var fileSystemWatcher = new Mock<IFileSystemWatcher>();
+
+            var fileSystemWatcherFactory = new Mock<IFileSystemWatcherFactory>();
+            fileSystemWatcherFactory.Setup(f => f.FromPath("C:\\Some Path\\.git")).Returns(fileSystemWatcher.Object);
+
+            var testSubject = CreateTestSubject("C:\\Some Path", fileSystemWatcherFactory);
+
+            fileSystemWatcher.VerifySet(w => w.Filter = "HEAD", Times.Once);
+            fileSystemWatcher.VerifySet(w => w.EnableRaisingEvents = true, Times.Once);
+
+            testSubject.Dispose();
+
+            fileSystemWatcher.Verify(w => w.Dispose(), Times.Once);
+        }
+
+        [TestMethod]
+        public void HeadChanged_NoListeners_DoNotThrow()
+        {
+            var fileSystemWatcher = new Mock<IFileSystemWatcher>();
+
+            var fileSystemWatcherFactory = new Mock<IFileSystemWatcherFactory>();
+            fileSystemWatcherFactory.Setup(f => f.FromPath("C:\\Some Path\\.git")).Returns(fileSystemWatcher.Object);
+
+            _ = CreateTestSubject("C:\\Some Path", fileSystemWatcherFactory);
+
+            fileSystemWatcher.VerifySet(w => w.Filter = "HEAD", Times.Once);
+            fileSystemWatcher.VerifySet(w => w.EnableRaisingEvents = true, Times.Once);
+
+            fileSystemWatcher.Raise(w => w.Changed += null, null, null);
```

Comment:
There's a `Should().NotThrow(action)` FluentAssertion method that would make the check more explicit (we don't appear to have used it so far, but perhaps we should)